Shopify/tophat
# Display devices and simulators in separate sections

### What does this change accomplish?

This change updates how devices are displayed in the Tophat menu.  Previously, devices and pinned simulators specifically were displayed under a single "Devices" section, with all remaining simulators in an "Other Devices" section.

Now, devices get their own section, pinned simulators get their own section, and the remaining simulators are displayed in an "Other Simulators" section.  This should make the pinning mechanism slightly less confusing.

Also, the "Other Simulators" section changes names to "Simulators" when no simulators are pinned to avoid confusion.

### How have you achieved it?

By reorganizing the UI code that produces these menus, taking the opportunity to convert `DeviceManager` to use the new Observation framework rather than `ObservableObject`.

### How can the change be tested?

Open the Tophat menu and ensure that you can interact with devices and simulators as per usual, but that they are organized into separate sections.
